qiskit/pulse/schedule.py

```diff
@@ -773,63 +773,122 @@ def _update_parameter_table(self, schedule: 'Schedule'):
+            interp_method: Deprecated. This argument is used by the legacy pulse drawer.
+            scale: Deprecated. This argument is used by the legacy pulse drawer.
+            channel_scales: Deprecated. This argument is used by the legacy pulse drawer.
+            plot_all: Deprecated. This argument is used by the legacy pulse drawer.
+            plot_range: Deprecated. This argument is used by the legacy pulse drawer.
+            interactive: Deprecated. This argument is used by the legacy pulse drawer.
+            table: Deprecated. This argument is used by the legacy pulse drawer.
+            label: Deprecated. This argument is used by the legacy pulse drawer.
+            framechange: Deprecated. This argument is used by the legacy pulse drawer.
+            channels: Deprecated. This argument is used by the legacy pulse drawer.
+            show_framechange_channels: Deprecated. This argument is used by the legacy pulse drawer.
+            draw_title: Deprecated. This argument is used by the legacy pulse drawer.
 
         Returns:
-            matplotlib.Figure: A matplotlib figure object of the pulse schedule.
+            Visualization output data.
+
+            The returned data type depends on the `plotter`.
+            If matplotlib family is specified, this will be a `matplotlib.pyplot.Figure` data.
+            The returned data is generated by the `.get_image` method of the specified plotter API.
         """
-        # pylint: disable=invalid-name, cyclic-import
-        from qiskit import visualization
-
-        return visualization.pulse_drawer(self, dt=dt, style=style,
-                                          filename=filename, interp_method=interp_method,
-                                          scale=scale, channel_scales=channel_scales,
-                                          plot_all=plot_all, plot_range=plot_range,
-                                          interactive=interactive, table=table, label=label,
-                                          framechange=framechange, channels=channels,
-                                          show_framechange_channels=show_framechange_channels,
-                                          draw_title=draw_title)
+        # pylint: disable=invalid-name, cyclic-import, missing-return-type-doc
+        from qiskit.visualization import pulse_drawer_v2
+
+        legacy_args = (dt, filename, interp_method, scale, channel_scales, plot_all,
+                       plot_range, interactive, table, label, framechange, channels,
+                       show_framechange_channels, draw_title)
+
+        if any(arg is not None for arg in legacy_args):
+            warnings.warn('Legacy pulse drawer is deprecated with some arguments. '
+                          'Please check the API document of new pulse drawer '
+                          '`qiskit.visualization.pulse_drawer_v2`.',
+                          DeprecationWarning)
```

Comment:
Should report what is deprecated for convenience.
```suggestion
        legacy_args = (dt, filename, interp_method, scale, channel_scales, plot_all,
                       plot_range, interactive, table, label, framechange, channels,
                       show_framechange_channels, draw_title)

        if any(arg is not None for arg in legacy_args):
            warnings.warn('Legacy pulse drawer is deprecated with some arguments. '
                          'Please check the API document of new pulse drawer '
                          '`qiskit.visualization.pulse_drawer_v2`.',
                          DeprecationWarning)
```